runtime/include/chpltypes.h

```diff
@@ -73,8 +73,7 @@ typedef _Bool chpl_bool;
 typedef bool chpl_bool;
 #endif
 
-static inline void* c_pointer_return(void* x) { return x; }
-static inline const void* c_pointer_return_const(const void* x) { return x; }
+static inline void* c_pointer_return(const void* x) { return (void*)x; }
```

Comment:
I think a comment here about why there's a const-dropping cast would be nice.

Reply:
Will do this as well as a TODO to eventually return constness, with explanation.